recipes/opengv/all/conanfile.py

```diff
@@ -0,0 +1,143 @@
+from conans import ConanFile, CMake, tools
+from conans.errors import ConanInvalidConfiguration
+import os
+import textwrap
+
+required_conan_version = ">=1.33.0"
+
+class opengvConan(ConanFile):
+    name = "opengv"
+    description = "A collection of computer vision methods for solving geometric vision problems"
+    url = "https://github.com/conan-io/conan-center-index"
+    homepage = "https://github.com/laurentkneip/opengv"
+    license = "BSD-3-Clause"
+    topics = ("computer", "vision", "geometric", "pose", "triangulation", "point-cloud")
+    exports_sources = "CMakeLists.txt"
+    generators = "cmake", "cmake_find_package"
+    settings = "os", "arch", "compiler", "build_type"
+    options = {
+        "shared": [True, False],
+        "fPIC": [True, False],
+        "with_python_bindings": [True, False]
+    }
+    default_options = {
+        "shared": False,
+        "fPIC": False,
+        "with_python_bindings": False
+    }
+
+    _cmake = None
+
+    @property
+    def _source_subfolder(self):
+        return "source_subfolder"
+
+    @property
+    def _module_subfolder(self):
+        return os.path.join("lib", "cmake")
+
+    @property
+    def _module_file_rel_path(self):
+        return os.path.join(self._module_subfolder,
+                            "conan-official-{}-targets.cmake".format(self.name))
+
+    @staticmethod
+    def _create_cmake_module_alias_targets(module_file, targets):
+        content = ""
+        for alias, aliased in targets.items():
+            content += textwrap.dedent("""\
+                if(TARGET {aliased} AND NOT TARGET {alias})
+                    add_library({alias} INTERFACE IMPORTED)
+                    set_property(TARGET {alias} PROPERTY INTERFACE_LINK_LIBRARIES {aliased})
+                endif()
+            """.format(alias=alias, aliased=aliased))
+        tools.save(module_file, content)
+
+    def validate(self):
+        # Disable windows builds since they error out.
+        if self.settings.os == "Windows":
+            raise ConanInvalidConfiguration("Windows builds are not supported by this recipe.")
+
+    def config_options(self):
+        if self.settings.os == "Windows":
```

Comment:
as this recipes doesn't work for windows, it is not needed, but also doesn't harm.

Reply:
I'd like to keep them to make the life of the developer that enables Windows build easier.
That might never happen tho.

Reply:
Yes, no problem.